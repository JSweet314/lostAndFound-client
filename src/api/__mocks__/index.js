export const reverseGeoCode = jest.fn().mockImplementation(() => ({
  results: [{formatted_address: 'Turing'}]
}));