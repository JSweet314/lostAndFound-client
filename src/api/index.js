export const addUser = async newUser => {
  try {
    const response = await fetch('/api/v1/users/new', {
      method: 'POST',
      body: JSON.stringify(newUser),
      headers: {
        'Content-Type': 'application/json'
      }
    });
    if (response.ok) {
      return await response.json();
    } else {
      throw new Error(`Errored with status code ${response.status}`);
    }
  } catch (error) {
    throw error;
  }
};