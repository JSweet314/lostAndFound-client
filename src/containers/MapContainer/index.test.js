import React from 'react';
import { shallow } from 'enzyme';
import { MapContainer } from './index';
import * as API from '../../api';

jest.mock('../../api');

describe('MapContainer', () => {
  let wrapper;
  const mockCaptureGeo = jest.fn();
  const mockHandleMapClick = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <MapContainer 
        captureGeo={mockCaptureGeo}
        loading={false}
        markerCoords={{}}
        routeId='map'
        handleMapClick={mockHandleMapClick}
        top=''
        right=''
        height=''
        width=''
      />
    );
  });
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able reverseGeoCode location info from a map', () => {
    wrapper.instance().reverseGeoCode({lat: 1, lng: 2});
    expect(API.reverseGeoCode).toHaveBeenCalledWith({lat: 1, lng: 2});
  });

  it('should call handleMapClick after reverseGeoCoding', async () => {
    wrapper.instance().reverseGeoCode({ lat: 1, lng: 2 });
    expect(mockHandleMapClick).toHaveBeenCalledWith({
      name: 'Turing', 
      lat: 1, 
      lng: 2
    });
  });

  it('should have a onMapClick method which calls reverseGeoCode', () => {
    const mockEvent = {
      latLng: {
        lat: jest.fn(() => 1),
        lng: jest.fn(() => 2)
      }
    };
    const mockReverseGeoCode = jest.fn();
    wrapper.instance().reverseGeoCode = mockReverseGeoCode;
    wrapper.instance().onMapClick(mockEvent);
    expect(mockReverseGeoCode).toHaveBeenCalledWith({lat: 1, lng: 2});
  });

  it('onMapClick should set the state with markerCoords', () => {
    const mockEvent = {
      latLng: {
        lat: jest.fn(() => 1),
        lng: jest.fn(() => 2)
      }
    };
    const mockReverseGeoCode = jest.fn();
    wrapper.instance().reverseGeoCode = mockReverseGeoCode;
    wrapper.instance().onMapClick(mockEvent);
    expect(wrapper.state('markerCoords')).toEqual({lat: 1, lng: 2});
  });
});