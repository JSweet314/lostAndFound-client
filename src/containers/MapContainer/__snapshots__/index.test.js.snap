// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapContainer
    captureGeo={[Function]}
    handleMapClick={[Function]}
    height=""
    loading={false}
    markerCoords={Object {}}
    right=""
    routeId="map"
    top=""
    width=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "center": null,
      "containerElement": <div
        className="lostAndFoundMap"
        style={
                Object {
                        "height": "50%",
                        "left": "",
                        "position": "absolute",
                        "top": "10rem",
                        "width": "50%",
                      }
        }
/>,
      "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyCVepTj3yZKdBV-t1k2wz5aHkbKYWFPL0w&v=3.exp&libraries=geometry,drawing,places",
      "loading": false,
      "loadingElement": <div
        style={
                Object {
                        "height": "100%",
                      }
        }
/>,
      "mapElement": <div
        style={
                Object {
                        "height": "100%",
                      }
        }
/>,
      "markerCoords": Object {},
      "onMapClick": [Function],
      "onMarkerClick": undefined,
      "position": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "center": null,
        "containerElement": <div
          className="lostAndFoundMap"
          style={
                    Object {
                              "height": "50%",
                              "left": "",
                              "position": "absolute",
                              "top": "10rem",
                              "width": "50%",
                            }
          }
/>,
        "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyCVepTj3yZKdBV-t1k2wz5aHkbKYWFPL0w&v=3.exp&libraries=geometry,drawing,places",
        "loading": false,
        "loadingElement": <div
          style={
                    Object {
                              "height": "100%",
                            }
          }
/>,
        "mapElement": <div
          style={
                    Object {
                              "height": "100%",
                            }
          }
/>,
        "markerCoords": Object {},
        "onMapClick": [Function],
        "onMarkerClick": undefined,
        "position": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
